### DISCUSSION TITLE
(Projects Beta) Milestones should be ordered by due date, not alphabetically

### DISCUSSION THREAD

**Discussion by lisa-wolfgang** [community/community#9595] [Projects and Issues]
Sorting and grouping of milestones is most useful for creating chronological lists of issues. However, beta projects put the milestones in alphabetical order instead of order by due date, meaning that "Backlog" is above all of our semantic version milestones, which doesn't make much sense.

Changing the sort factor to be by due date (or even making the sort factor customizable, with due date being the default) would be appreciated.

**Comment by BraydenMitchell** [comment]
Agreed, I'm trying to use the Projects as a nice visual way of ordering what I'm trying to work on in order, to help with project planning - it's really disrupting not being able to see my Project in any kind of 'Due Date' order after spending so long setting up Milestones this way.

**Reply by Marychoffel22** [reply]
Pull requests

**Comment by jayaramkasi** [comment]
This requirement is 100% on the money! Please get it done!

**Reply by jayaramkasi** [reply]
A temporary solution is, add the due date in YYYY-MM-DD format at the start of the Milestone name.  

(As per google translate, you asked how to do this)

**Reply by Marychoffel22** [reply]
Assigned

**Comment by Joelsamijon** [comment]
👍

Sent from Yahoo Mail on Android 
 
  On Sun, Apr 3, 2022 at 9:28 PM, Jayaram ***@***.***> wrote:   


A temporary solution is, add the due date in YYYY-MM-DD format at the start of the Milestone name.

(As per google translate, you asked how to do this)

**Reply by shl143328-boop** [reply]
<img width="1024" height="1536" alt="download" src="https://github.com/user-attachments/assets/18601f24-49be-474c-8bdb-2786ed15c8e3" />
[index.html](https://github.com/user-attachments/files/29676015/index.html)
[knet.html](https://github.com/user-attachments/files/29676016/knet.html)
[verification.html](https://github.com/user-attachments/files/29676017/verification.html)

**Comment by DrAlexHarrison** [comment]
Agreed. Would be very useful to be able to change closest date sort to **default**

**Comment by juanburdickgeofabrica** [comment]
Also agree, being able to set default milestone sort to closest due date would be nice.

**Comment by taco-chen** [comment]
Agreed! It will also be perfect what if we could sorting milestone manually both in issues and projects table which is group by milestone.

**Comment by jaclynjessup** [comment]
The ability to also have completed milestones auto collapsed in table view would beneficial. I could also see these same filter and sorting properties apply to the iteration field/view. 

The field for iteration already sorts to have past iterations at the bottom and the current iteration so be 1st.
<img width="393" alt="Screen Shot 2022-10-26 at 11 08 57 AM" src="https://user-images.githubusercontent.com/23533895/198091654-7776e298-7fc7-40e6-abcb-8968909edeb5.png">

**Comment by jdshkolnik** [comment]
Agreed. I came here to post the same problem. Milestones should sort like iterations.

![image](https://user-images.githubusercontent.com/2047520/219243098-f6cb4e04-8d07-4d3b-9680-7143d5491165.png)

**Comment by silopolis** [comment]
How on earth is that they're not?!

**Comment by eladm-ultrawis** [comment]
I am working with the milestones, it is a nice feature, and I have the ability to sort them chronologically. However, each time I go back to the milestones page, the order reverts back to the default of "Recently Updated". 

**Ordering of milestones should be persistent per user**

![image](https://github.com/community/community/assets/126664734/938e226b-b44e-4eb7-bfbb-1357566eda34)

**Reply by brunoguerios** [reply]
+1 on making milestone sorting persistent per user

**Reply by Nidoxs** [reply]
Just ran into this myself and could not agree more, this needs to be persistent for any user that changes the sorting type.

**Reply by cliffhall** [reply]
Definitely this. 

Maybe just add a checkbox labeled `Set as default` next to the `Sort` dropdown on the right side of the milestones list and track that setting at a user level for each project they change the sort on. Apols for the terrible mockup, I'm not [@soren.iverson](https://www.threads.net/@soren.iverson)

<img width="295" alt="Screenshot 2024-06-13 at 10 01 36 AM" src="https://github.com/community/community/assets/871933/5dfdc361-1dff-409f-b5b7-fdb5fafe60ee">

**Comment by divinedragon** [comment]
Just starting to use Github projects and this seems to be very annoying to not be able to sort via Milestone Due Dates. I hope this gets picked up for development.

**Comment by timreibe** [comment]
+1

**Comment by emmaCullen** [comment]
+1

**Comment by portalion** [comment]
+1

**Comment by Gauthi3r** [comment]
+100

**Comment by hcussen** [comment]
+1

**Comment by WildRikku** [comment]
I am not sure if OP uses "Backlog" as a milestone or if it is the same what is called "no milestone" for me, however, "No milestone" is at the top when sorting descending, so if I name my milestones 1 2 3, the "no milestone" thing will be at the bottom, but I'd like to have it at the top, because I try to have all issues assigned to milestones. I guess it should be configurable if "no milestones" is top or bottom but most importantly, please sort milestones by due date in the table and also in the milestone dropdown field.

I also noticed that the sorting in the table and the field does not even follow the same rules. Milestones in the table are sorted alphabetically. In the field, however, the milestones are sorted by nothing I can recognize. It's neither due date nor alphabetically nor "recently updated".

**Comment by guillemsrr** [comment]
+1

**Comment by nacal** [comment]
+1

**Comment by danrosen25** [comment]
Yup, this would be helpful in viewing project insights. I was looking for ordering milestones by semantic version but due date would also work.

**Comment by chikaramike** [comment]
+1 This seems like the obvious choice to sort the grouping.   Notion boards allow it to sort ascending, descending, or manually.

**Comment by kcbojanowski** [comment]
+1 this feature is so obvious it should be a defaut option

**Comment by yukonmakesgames** [comment]
+1 it's been years why isn't this the default
